# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaDecoderStateMachine.cpp
# Commit: c54cb3c10992
# Full Hash: c54cb3c109922bc15fb6d0ca67e5f2a05980c1a5
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-07-31 15:46:43
# Regressor Bug: 1493613
# File Overlap Count: 2
# Description:
#   Bug 1493613 - Move MediaStream control from DOMMediaStream to MediaStreamTrack. r=padenot
#   
#   This is inherently large, because modifying these bits of DOMMediaStream and
#   MediaStreamTrack affects all consumers and producers of all DOMMediaStreams and
#   MediaStreamTracks.
# ==============================================================================

diff -r 0b03dd9d20ac -r c54cb3c10992 dom/media/MediaDecoderStateMachine.cpp
--- a/dom/media/MediaDecoderStateMachine.cpp	Wed Jul 31 09:35:06 2019 +0000
+++ b/dom/media/MediaDecoderStateMachine.cpp	Wed Jul 31 07:58:17 2019 +0000
@@ -3339,7 +3339,6 @@
   MOZ_ASSERT(NS_IsMainThread());
   if (mOutputStreamManager) {
     mOutputStreamManager->Disconnect();
-    mNextOutputStreamTrackID = mOutputStreamManager->NextTrackID();
   }
   return InvokeAsync(OwnerThread(), this, __func__,
                      &MediaDecoderStateMachine::Shutdown);
@@ -3777,14 +3776,13 @@
 }
 
 void MediaDecoderStateMachine::EnsureOutputStreamManager(
-    MediaStreamGraph* aGraph) {
+    MediaStreamGraphImpl* aGraph) {
   MOZ_ASSERT(NS_IsMainThread());
   if (mOutputStreamManager) {
     return;
   }
-  mOutputStreamManager = new OutputStreamManager(
-      aGraph->CreateSourceStream(), mNextOutputStreamTrackID,
-      mOutputStreamPrincipal, mAbstractMainThread);
+  mOutputStreamManager = new OutputStreamManager(aGraph, mOutputStreamPrincipal,
+                                                 mAbstractMainThread);
 }
 
 void MediaDecoderStateMachine::EnsureOutputStreamManagerHasTracks(
@@ -3801,33 +3799,20 @@
   }
   if (aLoadedInfo.HasAudio()) {
     MOZ_ASSERT(!mOutputStreamManager->HasTrackType(MediaSegment::AUDIO));
-    mOutputStreamManager->AddTrack(MediaSegment::AUDIO);
-    LOG("Pre-created audio track with id %d",
-        mOutputStreamManager->GetLiveTrackIDFor(MediaSegment::AUDIO));
+    RefPtr<SourceMediaStream> dummy =
+        mOutputStreamManager->AddTrack(MediaSegment::AUDIO);
+    LOG("Pre-created audio track with underlying stream %p", dummy.get());
+    Unused << dummy;
   }
   if (aLoadedInfo.HasVideo()) {
     MOZ_ASSERT(!mOutputStreamManager->HasTrackType(MediaSegment::VIDEO));
-    mOutputStreamManager->AddTrack(MediaSegment::VIDEO);
-    LOG("Pre-created video track with id %d",
-        mOutputStreamManager->GetLiveTrackIDFor(MediaSegment::VIDEO));
+    RefPtr<SourceMediaStream> dummy =
+        mOutputStreamManager->AddTrack(MediaSegment::VIDEO);
+    LOG("Pre-created video track with underlying stream %p", dummy.get());
+    Unused << dummy;
   }
 }
 
-void MediaDecoderStateMachine::SetNextOutputStreamTrackID(
-    TrackID aNextTrackID) {
-  MOZ_ASSERT(NS_IsMainThread());
-  LOG("SetNextOutputStreamTrackID aNextTrackID=%d", aNextTrackID);
-  mNextOutputStreamTrackID = aNextTrackID;
-}
-
-TrackID MediaDecoderStateMachine::GetNextOutputStreamTrackID() {
-  MOZ_ASSERT(NS_IsMainThread());
-  if (mOutputStreamManager) {
-    return mOutputStreamManager->NextTrackID();
-  }
-  return mNextOutputStreamTrackID;
-}
-
 class VideoQueueMemoryFunctor : public nsDequeFunctor {
  public:
   VideoQueueMemoryFunctor() : mSize(0) {}